GVRf/Framework/framework/src/main/java/org/gearvrf/GVRMaterial.java

```diff
@@ -164,6 +164,21 @@ public GVRMaterial(GVRContext gvrContext) {
         this(gvrContext, GVRShaderType.Texture.ID);
     }
 
+    /**
+     * Create a new holder for a shader's uniforms from an existing material.
+     *
+     * @param src   GVRMaterial to copy from
+     * @param shaderId  shader ID
+      */
+    public GVRMaterial(GVRMaterial src, GVRShaderId shaderId)
+    {
+        super(src, shaderId);
+        if (hasTexture(mMainTextureName))
```

Comment:
This if looks to be redundant. mMainTextureName is going to be null at this time.